### ISSUE TITLE
HTTP BLAST parameters

### ISSUE THREAD

**Issue by 473021677** [DittmarLab/HGTector#22]
Hi Qiyunzhu:
    Thanks for your excellent software. I intend to use HGTector to infer putative HGTs. I have downloaded the packages for HGTector v0.2.2 and uncompressed it successfully. I have created the configuration file "config.txt" in the working directory. I want to call the NCBI server to perform BLAST and to retrieve sequence / taxonomy information. I want to modify the parameters(# Number of requests sent simultaneously:requests=1, # Time delay (sec) between requests:delay=30, # Time out (sec) for giving up a long wait:timeout=600) to accelerate the searching process. I intend to modify them as follows: # Number of requests sent simultaneously:requests=5, # Time delay (sec) between requests:delay=10, # Time out (sec) for giving up a long wait:timeout=100. I am not sure if it will influence the prediction accuracy for HGT genes. Would it be possible for you to help me? Thanks.

SIncerely,
Yuan

**Comment by qiyunzhu** [comment]
Hi @473021677 I hope you have figured it out already. But it seems that remote BLAST isn't efficient these days. Better do local search instead.

**Comment by 473021677** [comment]
Hi Qiyun:
    Thanks for your reply very much. HGTector is capable of connecting to the NCBI BLAST server to perform the search. But this has indeed become a time inefficienct solution recently. I want to do remote BLAST and local search and compare the results. I have tried do remote BLAST but the program terminated prematurely. I have appended the config.txt and log.txt. Would it be possible for you to help me solve this problem. Thanks.

Best regards,
Yuan
[config.txt](https://github.com/DittmarLab/HGTector/files/3129740/config.txt)
[FK2_LJY.Bin124.log](https://github.com/DittmarLab/HGTector/files/3129741/FK2_LJY.Bin124.log)

**Comment by qiyunzhu** [comment]
Hello Yuan, Remote BLAST is quite inefficient since NCBI does not allow frequent connections. If you have a third-party BLAST server that would be better. If you really want to try remote BLAST, please use the code attached here. These are part of the current development version of HGTector (0.3.0), which I haven't finished, but the BLAST part is completed. It should support all modes of BLAST better, including remote BLAST.

[HGTector-0.3.0-dev.zip](https://github.com/DittmarLab/HGTector/files/3129841/HGTector-0.3.0-dev.zip)

**Comment by 473021677** [comment]
Hi Qiyun:
   Thanks for your help very much. I have downloaded the software package and decompressed it. After I typed the command "perl /data/data/yuanyang/softwares/HGTector-0.3.0-dev/searcher.pl -h", the  error shows as follows "Error: Invalid configuration file: /data/data/yuanyang/softwares/HGTector-0.3.0-dev/../config.yml". Would it be possible for you to help me solve this problem. By the way, how I can create the text file "config.txt". Thanks very much. 

Best regards,
Yuan
